runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -1363,6 +1385,10 @@ private void commitLoop() {
   // Adds the commit to the commitStream if it fits, returning true iff it is consumed.
   private boolean addCommitToStream(Commit commit, CommitWorkStream commitStream) {
     Preconditions.checkNotNull(commit);
+    // Drop commits for failed work. Such commits will be dropped by Windmill anyway.
+    if (commit.work().isFailed()) {
+      return true;
```

Comment:
shoudl we count to show on status page?

Reply:
Are you thinking using something like the `pendingDeltaCounters` in this class?

Reply:
That ends up as a counter in dataflow backend. I was just thinking a local variable that is just rendered on the status page.  But also unnecessary if you want to just drop that request to make sure we make the cut.

Reply:
It seems like we'd want some kind of sliding window counter, e.g. how many failed in the last 5 minutes. Otherwise we just get some arbitrarily large number after the pipeline has been running for awhile. Let's wait on adding some visibility for now.